### ISSUE TITLE
Feature Request: an option to deal with duplicated header

### ISSUE THREAD

**Issue by ttyrueiwoqp** [tilo/smarter_csv#111]
Hi,

Thanks for the great library!

One issue we have been facing is that we have headers generated dynamically and there can be duplicates. In extensions/hash.rb, it assumes headers are unique by using `to_h`. Is it possible to add an option, e.g. `preserve_duplicated_headers`, which uses a function like below:
```
def self.zip_preserve_duplicated_keys(keys,values)
  hash = keys.zip(vals).group_by(&:first)
  hash.keys.each { |k| hash[k] = hash[k].map(&:last) }
  hash
end
```
[Here](https://stackoverflow.com/questions/30423632/preserving-key-value-pairs-for-duplicate-keys-when-merging-two-arrays-to-a-hash) is the reference.

In this way, cell values with duplicated headers will be stored in arrays instead of being overwritten, so we may iterate through the index to get them. You may need to deal with the array in the library as well.

**Comment by tilo** [comment]
Seems like a rare corner case to me. 
Shouldn't there be a warning instead, just informing you to fix your headers before importing?

**Comment by ttyrueiwoqp** [comment]
Yup it's not a common case. Currently the workaround that I can think of is, to postfix the header with an index while exporting to make headers unique, and then remove the index while importing. The tricky part is there's no way to distinguish whether the index is a postfix or a part of the original header, but I think I can live with that. Thanks!

**Comment by tilo** [comment]
this is addressed in 1.2.0 and in 2.0.0

**Comment by tilo** [comment]
to be clear - the change in 1.2.0 was only to throw an error
